### ISSUE TITLE
Inconsistent accuracy and loss in label predictor

### ISSUE THREAD

**Issue by sivannavis** [facebookresearch/sound-spaces#87]
Hi @yyf17 @ChanganVR  thanks for the discussion, hope you are doing good! I'm new to this repo and had some trouble training this label predictor with the given parameter settings and hope to get some suggestions. I'm in the same situation here with accuracy, it starts at a high value and decreases as the training loss decreases, trained from scratch. @yyf17 could you provide some tips about what the problem is?
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/50509896/185171471-96fe8c9d-4824-4b3a-b24c-7b7a2d090259.png">

Also about runtime, I'm wondering why the first epoch is so slow with 20% CPU util and 0% GPU util. In other epochs, CPU util stays around 100% but GPU util keeps jumping between 0 to 100%. I'm training via Slurm with 1 node, 1024 batch size, and 8 workers. I guess it has something to do with the CPU IO bottleneck in the data loader? Do you have any workaround advice? I really appreciate any help you can provide.

_Originally posted by @sivannavis in https://github.com/facebookresearch/sound-spaces/issues/55#issuecomment-1212628997_

**Comment by ChanganVR** [comment]
@sivannavis I think I pretrained the classifier and freezed it during navigation. The decrease of accuracy over time might have something to do with the non-stationary distribution of agent sampling actions and thus the distribution of classes. In another word, the accuracy measured in both training and validation stage might not be representative of all classes and thus it could be biased towards the high end initally. To verify this hypothesis, you could take the classifier trained after one epoch and evaluate it on the complete classification test set.

**Comment by sivannavis** [comment]
@ChanganVR Thank you for the reply! This is not in the navigation process yet, same as #55, only using spectrograms to predict categories, I don't think actions are involved in calculating the accuracy here. While I also find the `torch.abs()` in the audiogoal_trainer.py hard to understand: https://github.com/facebookresearch/sound-spaces/blob/fb68e410a4a1388e2d63279e6b92b6f082371fec/ss_baselines/savi/pretraining/audiogoal_trainer.py#L148
I'm not sure but it seems to make more sense to use `predicts` directly because the abs() gives the most/least possible answer without softmax.

Meanwhile, could you tell me if training savi with run.py is already using the pretrained label predictor, or do I need to specify it anywhere?

**Comment by sivannavis** [comment]
Oh, I think I found the answer to my second question in this issue https://github.com/facebookresearch/sound-spaces/issues/46#issuecomment-898688576.

I hope to confirm the following things:
1. I used `python ss_baselines/savi/pretraining/audiogoal_trainer.py --run-type train --model-dir /scratch/sd5397/soundspaces/data/models/savi3 --predict-label` to train the label predictor, which I assume is not using any pretrained predictor, and yet my accuracy starts at a very high point 95%, compared to the results from this issue.
2. As for the first step of training savi, the code uses the pretrained label predictor weights here: ` state_dict = torch.load('data/pretrained_weights/semantic_audionav/savi/label_predictor.pth') ` in belief_predictor. If I want to use my trained label predictor, I need to update this path with my previous best model in the output folder. Then in this process, the label predictor is fixed, and location predictor is trained.
3. As for the second step of training savi, I need to check the output folder from the first step and update this line with the best model https://github.com/facebookresearch/sound-spaces/blob/fb68e410a4a1388e2d63279e6b92b6f082371fec/ss_baselines/savi/config/semantic_audionav/savi.yaml#L60 . Then in this process, the entire goal descriptor is fixed and all other networks are trained.

Could you help me clarify if my understanding is right? Thank you a lot!

**Comment by ChanganVR** [comment]
@sivannavis please see my responses to you below:

1. Yes. Apparently, the classification task is pretty easy and that's why the lable prediction has a very high accuracy after just one epoch of training.
2. Yes. 
3. Yes, the pretrained weights in the first step goes here.
